### DISCUSSION TITLE
Is the direction flipped in the code?

### DISCUSSION THREAD

**Discussion by sanandnarayan** [Jeiwan/uniswapv3-book#46] [Milestone 2. Second Swap]
In Tick Bitmap index chapter it is mentioned that
"lte is the flag that sets the direction. When true, we’re selling token x and searching for next initialized tick to the right of the current one"

When I run the testcase testBuyUSDCConsecutivePriceRanges. I am selling ETH i.e selling token X. And lte is true.
<img width="833" alt="Screen Shot 2022-12-21 at 4 37 10 PM" src="https://user-images.githubusercontent.com/433334/209029905-274c9232-b55f-4afb-bda9-8383ee973207.png">
The above subtracts. So we look for lower prices than current. So we look left and not right of the current tick?

Please help me understand why you mention we look to the right of the current one for the next initialized tick?

**Comment by Jeiwan** [comment]
This comes from how we usually count bits in binary numbers: from right to left. So, in a word, the highest price is at the highest (the leftmost) bit and the lowest price is at the lowest (the rightmost) bit. Words are traversed from left to right when price grows and from right to left when price falls, but bits inside a word are always ordered from right to left.

This is how Uniswap does it, so I decided to leave it as it is, even though it might look confusing:
https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/TickBitmap.sol#L39
https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/TickBitmap.sol#L51-L53
https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/TickBitmap.sol#L66
